File: Sponsors/mw-streams-java/src/main/java/com/mwam/hackathon/sources/IterSource.java
package com.mwam.hackathon.sources;

import java.util.ArrayList;

public class IterSource<T> extends Source<T> {
    private final Iterable<T> items;

    public IterSource(Iterable<T> items) {
        super(new ArrayList<>());
        this.items = items;
    }

    @Override
    public void start() {
        for (final var item : items) {
            emit(item);
        }

        flush();
    }
}

File: Sponsors/mw_streams_python/mws/sources/iter_source.py
from typing import TypeVar, Iterable

from mws.operators.operator import Source


T = TypeVar("T")


class IterSource(Source[T]):
    def __init__(self, items: Iterable[T]) -> None:
        super().__init__()
        self.items = items

    def start(self) -> None:
        for item in self.items:
            self.emit(item)

        #
        self.flush()
